collections/sum_of_test.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../testing/asserts.ts";
+import { sumOf } from "./sum_of.ts";
+
+Deno.test("[collections/sumOf] On object properties", () => {
+  const object = [
+    { name: "Kyle", age: 34 },
+    { name: "John", age: 42 },
+    { name: "Anna", age: 23 },
+  ];
+
+  const actual = sumOf(object, (i) => i.age);
```

Comment:
Sum of negatives and positives, multiple negatives, etc?

Reply:
Added some more tests related to negatives, thanks!